include/nil/actor/math/polynomial/polynomial_view.hpp

```diff
@@ -396,15 +396,20 @@ namespace nil {
                         if (other.size() + shift + 1 > r.size()) {
                             r.resize(other.size() + shift + 1);
                         }
-                        auto glambda = [=](const FieldValueType& x, const FieldValueType& y) {
-                            return y - (x * lead_coeff);
-                        };
-                        std::transform(other.begin(), other.end(), r.begin() + shift, r.begin() + shift, glambda);
+                        detail::block_execution(
+                            this->size(),
+                            smp::count,
+                            [this, lead_coeff, shift](std::size_t begin, std::size_t end) {
+                                for (std::size_t i = begin; i < end; i++) {
+                                    this[i + shift] -= other[i] * lead_coeff;
+                                }
+                            }
+                        );
                         r.condense();
 
                         r_deg = r.size() - 1;
                     }
-                    //                    q.condense();
```

Comment:
From my opinion - if we write algorithm better, we don't need condense for q. 
For some reason we construct q as 
```
polynomial_view r(*this);
container_type q(r.size(), FieldValueType::zero());
```
But we now, that q size must be this.size() - other.size() (as maximal degree of division result). 
This mark important because we can improve algorithm speed and memory consumption.

Reply:
This one must be done in crypto3 first, then moved to actor.